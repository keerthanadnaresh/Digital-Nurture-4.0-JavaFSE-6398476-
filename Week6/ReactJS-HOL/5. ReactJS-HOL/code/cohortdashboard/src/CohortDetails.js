import React from 'react';
import styles from './CohortDetails.module.css';

function CohortDetails({ cohort }) {
  const colorStyle = {
    color: cohort.status === 'Ongoing' ? 'green' : 'blue',
    fontWeight: 'bold',
  };

  return (
    <div className={styles.box}>
      <h4 style={colorStyle}>{cohort.name}</h4>
      <dl>
        <dt>Started On</dt>
        <dd>{cohort.startDate}</dd>
        <dt>Current Status</dt>
        <dd>{cohort.status}</dd>
        <dt>Coach</dt>
        <dd>{cohort.coach}</dd>
        <dt>Trainer</dt>
        <dd>{cohort.trainer}</dd>
      </dl>
    </div>
  );
}

export default CohortDetails;
