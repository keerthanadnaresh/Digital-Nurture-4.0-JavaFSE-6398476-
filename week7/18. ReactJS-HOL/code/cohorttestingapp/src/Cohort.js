export const CohortData = [
  {
    id: 1,
    name: 'Cohort Alpha',
    technology: 'ReactJS',
    status: 'ongoing'
  },
  {
    id: 2,
    name: 'Cohort Beta',
    technology: 'Angular',
    status: 'completed'
  }
];
