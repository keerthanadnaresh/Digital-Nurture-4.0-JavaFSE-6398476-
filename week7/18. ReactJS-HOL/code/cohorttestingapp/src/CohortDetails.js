import React from 'react';

const CohortDetails = ({ cohort }) => {
  return (
    <div>
      <h2>{cohort.name}</h2>
      <p>Technology: {cohort.technology}</p>
      <p>Status: {cohort.status}</p>
    </div>
  );
};

export default CohortDetails;
