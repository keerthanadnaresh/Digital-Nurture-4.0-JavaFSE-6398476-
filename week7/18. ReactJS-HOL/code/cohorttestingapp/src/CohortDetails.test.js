import { render, screen } from '@testing-library/react';
import CohortDetails from './CohortDetails';
import { CohortData } from './Cohort';

describe("CohortDetails component", () => {
  test("should show technology field", () => {
    render(<CohortDetails cohort={CohortData[0]} />);
    const techElement = screen.getByText(/Technology:/i);  // match label
    expect(techElement).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<CohortDetails cohort={CohortData[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
