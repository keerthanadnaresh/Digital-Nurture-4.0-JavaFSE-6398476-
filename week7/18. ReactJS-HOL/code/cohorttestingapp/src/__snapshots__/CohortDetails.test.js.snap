// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortDetails component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Cohort Alpha
    </h2>
    <p>
      Technology: ReactJS
    </p>
    <p>
      Status: ongoing
    </p>
  </div>
</DocumentFragment>
`;
